src/Symfony/Bundle/WebServerBundle/DependencyInjection/WebServerExtension.php

```diff
@@ -54,4 +60,16 @@ private function getPublicDirectory(ContainerBuilder $container)
 
         return $kernelProjectDir.'/'.$publicDir;
     }
+
+    private function getPidFileDirectory(ContainerBuilder $container)
+    {
+        $kernelCacheDir = $container->getParameter('kernel.cache_dir');
+        $environment = $container->getParameter('kernel.environment');
+
+        if (basename($kernelCacheDir) !== $environment) {
+            return $container->getParameter('kernel.project_dir');
+        }
+
+        return \dirname($container->getParameter('kernel.cache_dir'), 1);
```

Comment:
And otherwise use the root cache folder. @nicolas-grekas WDYT?

Reply:
`1` already is the default argument value. Passing it here is not necessary.

Reply:
By the way, not sure it's a good idea to use the parent directory of the cache directory. It may not be writable.